From c1c793b570e94b8913d0a97bf3c09d842605cce1 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 3 Dec 2018 19:09:11 +0200
Subject: [PATCH 0226/1239] arm64: mvebu: dts: a37xx: add DDR4 db-C rev 3.x dts

This patch added dts support for DDR4 with rev 3.x ECO.
Config "C" dts configurates comphy0 with sgmii1-sfp2.5g.

Change-Id: I7766739b112f0fe09735157b419670758c298e0c
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
---
 arch/arm/dts/Makefile                     |  1 +
 arch/arm/dts/armada-3720-ddr4-db-v3-C.dts | 63 +++++++++++++++++++++++
 2 files changed, 64 insertions(+)
 create mode 100644 arch/arm/dts/armada-3720-ddr4-db-v3-C.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index cf65d8c65d..4d811f1945 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -178,6 +178,7 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-3720-ddr3-db-v1-C.dtb		\
 	armada-3720-ddr4-db-v3-A.dtb		\
 	armada-3720-ddr4-db-v3-B.dtb		\
+	armada-3720-ddr4-db-v3-C.dtb		\
 	armada-3720-ddr4-db-v1-A.dtb		\
 	armada-3720-ddr4-db-v1-B.dtb		\
 	armada-3720-espressobin.dtb		\
diff --git a/arch/arm/dts/armada-3720-ddr4-db-v3-C.dts b/arch/arm/dts/armada-3720-ddr4-db-v3-C.dts
new file mode 100644
index 0000000000..e4ae11c6a4
--- /dev/null
+++ b/arch/arm/dts/armada-3720-ddr4-db-v3-C.dts
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2018 Marvell International Ltd.
+ */
+
+#include "armada-3720-ddr4-db-v3-A.dts"
+
+/* detail board setup:
+ * Pcie Jumper slot	:	DB-88F3720-PCIe-mPCIe-Jumper
+ * SGMII module slot	:	DB-88F3720-SerDes-Jumper
+ * RGMII module slot	:	DB-88F3720-PHY module
+ * eMMC slot		:	DB-88F3720-eMMC_NAND module
+ * SDIO slot		:	DB-88F3720-SDcard module
+ *
+ * PHY0 (USB3/SGMII1)	:	SGMII1 via SFP2.5G (1 Lane)
+ * PHY1 (PCIe/SGMII0)	:	PCIe (WiFi via mini_PCIe module)
+ * PHY2 (SATA)		:	SATA
+ * RGMII-0		:	Ethernet via PHY (1 Lane)
+ * USB2 Port 0		:	USB2 as part USB3
+ * USB2 Port 1		:	USB2 Host
+ * eMMC*		:	eMMC NAND,
+				connected to SD / eMMC Host 1 in North Bridge
+ * SDIO*		:	SD card,
+				connected to SDIO Host 0 in South Bridge
+ */
+
+/ {
+	model = "Marvell Armada 3720 Development Board DB-88F3720-DDR4-V3";
+	compatible = "marvell,armada-3720-db-v3",
+		     "marvell,armada3720",
+		     "marvell,armada3710";
+
+	soc {
+		internal-regs {
+			comphy: comphy@18300 {
+				phy0 {
+					phy-type = <COMPHY_TYPE_SGMII1>;
+					phy-speed = <COMPHY_SPEED_3_125G>;
+				};
+			};
+			eth1: neta@40000 {
+				status = "okay";
+				phy-mode = "sgmii";
+				fixed-link {
+					/* Here "speed" is set to 1000, GBE MAC
+					 * is running in 1G mode, but the
+					 * actuall PHY speed may be 1 Gbps or
+					 * 2.5 Gbps, it's up to the
+					 * corresponding SERDES speed in comphy
+					 * node.
+					 * If SERDES speed is set to 3.125G, it
+					 * implies sgmii 2.5 Gbps;
+					 * if SERDES speed is set to 1.25G, it
+					 * implies sgmii 1 Gbps.
+					 */
+					speed = <1000>;
+					full-duplex;
+				};
+			};
+		};
+	};
+
+};
-- 
2.29.0

